error: cannot find type `MODE2_ALARM` in this scope

The library generated from a SAM L11 svd fails with the following error:
```
error[E0412]: cannot find type `MODE2_ALARM` in this scope
   --> src/rtc/mod.rs:125:23
    |
125 |     pub mode2_alarm0: MODE2_ALARM,
    |                       ^^^^^^^^^^^ not found in this scope
help: possible candidate is found in another module, you can import it into scope
    |
1   | use rtc::mode2::MODE2_ALARM;
    |
```

Somehow this field in the MODE2 register block fails to reference the `mode2` module like all the others do: 
```
    #[doc = "0x18 - MODE2 Clock Value"]
    pub clock: self::mode2::CLOCK,
    _reserved2: [u8; 4usize],
    #[doc = "0x20 - MODE2 Alarm"]
    pub mode2_alarm0: MODE2_ALARM,
    _reserved3: [u8; 27usize],
    #[doc = "0x40 - General Purpose"]
    pub gp: [self::mode2::GP; 2],
    _reserved4: [u8; 24usize],
```

I noticed that in the svd this seems to be defined as a cluster, rather than a register. Does this help explain the issue?

I am able to workaround through patches.

You can find the svd I used here - https://github.com/evq/atsaml11xxx/blob/master/ATSAML11E16A.orig.svd

--- a/src/generate/peripheral.rs
+++ b/src/generate/peripheral.rs
@@ -633,7 +633,7 @@ fn expand_cluster(
 
     match cluster {
         Cluster::Single(info) => cluster_expanded.push(RegisterBlockField {
-            field: convert_svd_cluster(cluster)?,
+            field: convert_svd_cluster(cluster, name)?,
             description: info.description.as_ref().unwrap_or(&info.name).into(),
             offset: info.address_offset,
             size: cluster_size,
@@ -653,7 +653,7 @@ fn expand_cluster(
 
             if array_convertible {
                 cluster_expanded.push(RegisterBlockField {
-                    field: convert_svd_cluster(&cluster)?,
+                    field: convert_svd_cluster(&cluster, name)?,
                     description: info.description.as_ref().unwrap_or(&info.name).into(),
                     offset: info.address_offset,
                     size: cluster_size * array_info.dim,
@@ -855,7 +855,7 @@ fn expand_svd_cluster(
     let mut out = vec![];
 
     match &cluster {
-        Cluster::Single(_info) => out.push(convert_svd_cluster(cluster)?),
+        Cluster::Single(_info) => out.push(convert_svd_cluster(cluster, name)?),
         Cluster::Array(info, array_info) => {
             let indices = array_info
                 .dim_index
@@ -884,14 +884,13 @@ fn expand_svd_cluster(
 }
 
 /// Convert a parsed `Cluster` into its `Field` equivalent
-fn convert_svd_cluster(cluster: &Cluster) -> Result<syn::Field, syn::Error> {
+fn convert_svd_cluster(cluster: &Cluster, name: Option<&str>) -> Result<syn::Field, syn::Error> {
     Ok(match cluster {
-        Cluster::Single(info) => new_syn_field(
-            &info.name.to_sanitized_snake_case(),
-            syn::Type::Path(parse_str::<syn::TypePath>(
-                &info.name.to_sanitized_upper_case(),
-            )?),
-        ),
+        Cluster::Single(info) => {
+            let ty_name = util::replace_suffix(&info.name, "");
+            let ty = name_to_ty(&ty_name, name)?;
+            new_syn_field(&info.name.to_sanitized_snake_case(), ty)
+        }
         Cluster::Array(info, array_info) => {
             let name = util::replace_suffix(&info.name, "");
 

